main.py

```diff
@@ -280,7 +281,14 @@ def get_switch_metadata(self, dpid):
     @rest('v3/switches/<dpid>/metadata', methods=['POST'])
     def add_switch_metadata(self, dpid):
         """Add metadata to a switch."""
-        metadata = request.get_json()
+        try:
+            metadata = request.get_json()
+        except BadRequest:
+            return jsonify('The request body is not a well-formed JSON.'), 400
+        if metadata is None:
+            return jsonify('The request body mimetype is not'
+                           ' application/json.'), 400
```

Comment:
If the metadata is empty, the error will always be the mime type? Is this right?

Reply:
I added a check to the empty request case(https://github.com/kytos/topology/pull/148/commits/0271d10862ba8934a35442e5427074298ead1b02), more information related to the content type error was also added.